src/main/java/org/jenkinsci/plugins/workflow/cps/steps/LoadFromClasspathStepExecution.java

```diff
@@ -0,0 +1,134 @@
+package org.jenkinsci.plugins.workflow.cps.steps;
+
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Scanner;
+
+import org.apache.tools.ant.taskdefs.Classloader;
+import org.codehaus.groovy.control.MultipleCompilationErrorsException;
+import org.jenkinsci.plugins.workflow.cps.CpsCompilationErrorsException;
+import org.jenkinsci.plugins.workflow.cps.CpsFlowExecution;
+import org.jenkinsci.plugins.workflow.cps.CpsStepContext;
+import org.jenkinsci.plugins.workflow.cps.CpsThread;
+import org.jenkinsci.plugins.workflow.cps.replay.ReplayAction;
+import org.jenkinsci.plugins.workflow.steps.BodyExecutionCallback;
+import org.jenkinsci.plugins.workflow.steps.StepContext;
+import org.jenkinsci.plugins.workflow.steps.StepExecution;
+
+import groovy.lang.Script;
+import hudson.model.TaskListener;
+
+/**
+ * Execution implementation for {@link LoadFromClasspathStep}.
+ *
+ * @author Austin Witt
+ * 
+ * @since 2.74
+ *
+ */
+public class LoadFromClasspathStepExecution extends StepExecution {
+	
+	private static final long serialVersionUID = 1L;
+
+	protected String path;
+
+	protected Object classloaderSource;
+	
+	protected boolean unexport;
+	
+	protected transient TaskListener listener;
+
+	public LoadFromClasspathStepExecution(
+		StepContext _context, 
+		String _path, 
+		Object _classloaderSource,
+		boolean _unexport)
+			throws
+				IOException,
+				InterruptedException
+	{
+		super( _context );
+
+		path = _path;
+		classloaderSource = _classloaderSource;
+		listener = _context.get( TaskListener.class );
+	}
+
+	@Override
+	public boolean start() throws Exception {
+		
+		CpsStepContext cps = (CpsStepContext) getContext();
+		CpsThread t = CpsThread.current();
+		CpsFlowExecution execution = t.getExecution();
+
+		String text = get_classpath_resource_contents( classloaderSource != null ? classloaderSource : this, path );
+		String clazz = execution.getNextScriptName( path );
+		String newText = ReplayAction.replace( execution, clazz );
+		if( newText != null ) {
+			listener.getLogger().println( "Replacing Groovy text with edited version" );
+			text = newText;
+		}
+
+		Script script;
+		try {
+			script = execution.getShell().parse(text);
+		} catch (MultipleCompilationErrorsException e) {
+			// Convert to a serializable exception, see JENKINS-40109.
+			throw new CpsCompilationErrorsException(e);
+		}
+
+		// execute body as another thread that shares the same head as this thread
+		// as the body can pause.
+		cps.newBodyInvoker(t.getGroup().export( script ), unexport )
```

Comment:
Based on the contents of `LoadStepExecution`, I guessed that `unexport: true` was the value I should use here.

However, I found the following issue:

# `@groovy.transform.Field` not available after pipeline restart

`Jenkinsfile`
```groovy
@groovy.transform.Field
def FOO=null

FOO="bar"

loadFromClasspath("com/example/Pipeline.groovy")
```

`Pipeline.groovy`
```
echo "FOO=${FOO}"
checkpoint()
echo "FOO="${FOO}"
```

This pipeline runs, but after resuming at the `checkpoint()`, property `FOO` is undefined. Switching to `unexport: false` fixed this.

# Something Else I didn't Pin Down

There was another issue which may have just been that `GlobalVariables` that got loaded during a `loadFromClasspath()`'d script weren't available after pipeline restart, either. However, after determining that `unexport: false` fixed the first issue, I checked and it also fixed the other issue I was seeing. I'm postponing further voodoo troubleshooting until I have a better understanding of `unexport`.